metric_learn/_util.py

```diff
@@ -405,3 +405,15 @@ def validate_vector(u, dtype=None):
   if u.ndim > 1:
     raise ValueError("Input vector should be 1-D.")
   return u
+
+
+def _check_num_dims(n_features, num_dims):
+  """Checks that num_dims is less that n_features and deal with the None
+  case"""
+  if num_dims is None:
+    dim = n_features

```

Comment:
As a style nitpick, I prefer early-return for cases like this:

```python
if num_dims is None:
  return n_features
if 0 < num_dims <= n_features:
  return num_dims
raise ValueError(...)
```

Reply:
That's seems better I agree, done